docs/docs/genai/prompt-version-mgmt/version-tracking/track-application-versions-with-mlflow.mdx

```diff
@@ -3,249 +3,373 @@ title: Track Application Versions with MLflow
-    os.environ["DATABRICKS_TOKEN"] = "your-databricks-token"  # Your access token
-    ```
-
-    ### Configure API Keys
-
-    Set your LLM provider API keys as environment variables:
-
-    ```python
-    import os
+    # Authentication (only needed outside Databricks notebooks)
+    os.environ["DATABRICKS_HOST"] = "https://your-workspace.databricks.com"
+    os.environ["DATABRICKS_TOKEN"] = "your-databricks-token"
 
-    # Set your OpenAI API key (required for this guide)
+    # Set LLM provider API key
     os.environ["OPENAI_API_KEY"] = "your-api-key-here"
-    ```
-
-    :::tip
-    You can also set these environment variables in your shell before running your script:
-
-    ```bash
-    # For Databricks authentication (if running outside Databricks)
-    export DATABRICKS_HOST="https://your-workspace.databricks.com"
-    export DATABRICKS_TOKEN="your-databricks-token"
-
-    # For LLM providers
-    export OPENAI_API_KEY="your-api-key-here"
-    ```
 
-    :::
-
-    ### Configure MLflow Tracking
-
-    Set up MLflow to connect to your tracking server and experiment:
-
-    ```python
-    import mlflow
-
-    # Set MLflow tracking URI and experiment
+    # Configure tracking
     mlflow.set_tracking_uri("http://127.0.0.1:5000")
     mlflow.set_experiment("my_genai_app_versions")
     ```
 
   </TabItem>
 </Tabs>
 
-## Step 1: Create a <APILink fn="mlflow.entities.LoggedModel">LoggedModel</APILink> (Metadata Hub)
+### Step 2: Create Version-Controlled Application
 
-A `LoggedModel` version serves as a central record (metadata hub) for a specific iteration of your application. It doesn't need to store the application code itself; instead, it points to where your code is managed (e.g., a Git commit hash).
-
-We use `mlflow.set_active_model()` to declare which `LoggedModel` we are currently working with, or to create a new one if it doesn't already exist. This function returns an `ActiveModel` object containing the `model_id` which is useful for subsequent operations.
+Link your application to git commits for complete traceability:
 
 ```python
-import subprocess
-
-# Define your application and its version identifier
-app_name = "customer_support_agent"
-
-# Get current git commit hash for versioning
-try:
-    git_commit = (
-        subprocess.check_output(["git", "rev-parse", "HEAD"])
-        .decode("ascii")
-        .strip()[:8]
-    )
-    version_identifier = f"git-{git_commit}"
-except subprocess.CalledProcessError:
-    version_identifier = "local-dev"  # Fallback if not in a git repo
-
-logged_model_name = f"{app_name}-{version_identifier}"
+import mlflow
+import openai
 
-# Set the active model context
-active_model_info = mlflow.set_active_model(name=logged_model_name)
-print(
-    f"Active LoggedModel: '{active_model_info.name}', Model ID: '{active_model_info.model_id}'"
-)
-```
+# Get current git commit using MLflow's built-in utilities
+from mlflow.utils.git_utils import get_git_commit
```

Comment:
I don't think we should expose `mlflow.utils.git_utils`. It's undocumented.

Reply:
Yep agreed. I removed the whole advanced example since it's a bit distracting and will be provided as a complete solution once your feature for git integration is done.